### ISSUE TITLE
Reorganize directory structure 

### ISSUE THREAD

**Issue by andreas-bok-sociomantic** [sociomantic/libdrizzle-redux#95]
The current project structure is shown below: 

```sh
./                Makefile and configure scripts. 
./deb             deb packaging configuration and generated deb packages
./docs            Documentation 
./libdrizzle      General sources
./libdrizzle-5.1  Header files that expose the public interface and are to be installed
./m4              autoconf m4 macros 
./rpm             rpm packaging configuration 
./tests           tests and test binaries 
./yatl            aux library for tests 
```

The directory structure should be changed to a more general layout. Some concerns are listed below:

- General sources and headers for the public interface should be located in `src` and `include` respectively. 
- A top level directory where packaging configuration, currently `deb` and `rpm`. 
- A top level `build` directory which contains all object files, and is removed on a clean.

See also: http://hiltmon.com/blog/2013/07/03/a-simple-c-plus-plus-project-structure/

Another option would be to restructure the build process to use another automated build system, e.g. `gyp, or meson` with `ninja` or `craftr` or `makeit`

**Comment by andreas-bok-sociomantic** [comment]
I have looked into `meson` and `ninja` and `cmake`. Given the de facto standard of `cmake` I will opt that tool.

**Comment by andreas-bok-sociomantic** [comment]
This issue is addressed in #116

**Comment by andreas-bok-sociomantic** [comment]
Closing this issue and opening a separate one #135 to address 

> A top level `build` directory which contains all object files, and is removed on a clean.
